code/game/objects/structures/coathanger.dm

```diff
@@ -1,51 +1,42 @@
 /obj/structure/coatrack
 	name = "coat rack"
-	desc = "Rack that holds coats."
+	desc = "Its a rack that can hold coats, and hats!"
 	icon = 'icons/obj/coatrack.dmi'
-	icon_state = "coatrack0"
+	icon_state = "coatrack"
 	var/obj/item/clothing/suit/coat
-	var/list/allowed = list(/obj/item/clothing/suit/storage/toggle/labcoat, /obj/item/clothing/suit/storage/det_trench)
+	var/obj/item/clothing/head/hat
 
 /obj/structure/coatrack/attack_hand(mob/user as mob)
-	user.visible_message("[user] takes [coat] off \the [src].", "You take [coat] off the \the [src]")
-	if(!user.put_in_active_hand(coat))
-		coat.dropInto(user.loc)
-	coat = null
-	update_icon()
+	if(coat)
+		user.visible_message(SPAN_NOTICE("[user] takes [coat] off \the [src]."), SPAN_NOTICE("You take [coat] off the \the [src]"))
+		if(!user.put_in_active_hand(coat))
+			coat.dropInto(user.loc)
+		coat = null
+		update_icon()
+	else if(hat)
+		user.visible_message(SPAN_NOTICE("[user] takes [hat] off \the [src]."), SPAN_NOTICE("You take [hat] off the \the [src]"))
+		if(!user.put_in_active_hand(hat))
+			hat.dropInto(user.loc)
+		hat = null
```

Comment:
update_icon can be moved from either if-else section to one indent back, as it happens in both.
Additionally, there could be a neat QOL feature of making it pick hat or coat depending on targeting limb. Not necessary, but would be really cool.

Reply:
that QOL feature is so based im doing that

Reply:
> update_icon can be moved from either if-else section to one indent back, as it happens in both. Additionally, there could be a neat QOL feature of making it pick hat or coat depending on targeting limb. Not necessary, but would be really cool.

that sounds absolutely awful, if anything we should be phasing out targeting limb as much as possible. just do a radial menu, people will 100% prefer that over having to change their fucking limb target.

Reply:
Radial menu takes more time from taking stuff off of it. With limb targeting it's faster + you can still just take random thing off if you didn't target appropriate limb.

Reply:
> Radial menu takes more time from taking stuff off of it. With limb targeting it's faster + you can still just take random thing off if you didn't target appropriate limb.

>radial menu is slow
>limb targeting is fast

what alternate dimension are you from

Reply:
resolution: limb targeting beter